### ISSUE TITLE
[FEATURE] add rabbit container to tests, or mock the calls

### ISSUE THREAD

**Issue by p-mcgowan** [acrontum/q-wrapper#5]
**Is your feature request related to a problem? Please describe.**
ci is failing because the rabbit host is unreachable

**Describe the solution you'd like**
Either add the container to the ci (not 100% sure on travis) or just mock the calls (less ideal, since the tests are checking if the connection / channels are closing properly)

**Additional context**
I'll look into it when I have some downtime, but if you guys have thoughts or better ideas let me know

**Comment by Acrontum-Carmichael** [comment]
Yea i saw this one too, kind of a tough one to test reliably without a service to connect to. A rabbit in a box would be cool for this.. if travis can handle it

**Comment by p-mcgowan** [comment]
yeah - it could be possible to use github actions with a rabbit image if travis can't or we could conditionally skip the tests depending on env, so that you could just force them to run when testing locally.

so by default the tests which run against external dependencies could be skipped, but running `CI=false npm test` (for example) would allow external service tests to run. 

it's always a bit of a pain to do this unless the CI env is set up to match a "typical" runtime env, which is generally not trivial
